byoqm/runner/runner.py

```diff
@@ -102,6 +107,14 @@ def _run_metrics(self) -> Dict:
         return results
 
     def _generate_violations_table(self, results: Dict, time: str):
+        """
+        _generate_violations_table generates a csv file in the output/violations/ folder containing
+        the locations of all found violations during parsing of the code base
+
+        _generate_violations_table generates given a dictionary of metrics, constructs a list containing
```

Comment:
The middle step adds confusion without clarification (it makes a list to then make a pandas dataframe). To me the important aspect here is I give a dict and I get a file